internal/examples/musicdb/database_test.go

```diff
@@ -206,11 +220,15 @@ func TestSingersAndAlbumsReadTransaction(t *testing.T) {
 			singersAndAlbums[3].Key(): singersAndAlbums[3],
 			singersAndAlbums[5].Key(): singersAndAlbums[5],
 		}
-		actual, err := musicdb.Query(client.Single()).BatchGetSingersRowsInterleaved(fx.Ctx, []musicdb.SingersKey{
-			singersAndAlbums[1].Key(),
-			singersAndAlbums[3].Key(),
-			singersAndAlbums[5].Key(),
-			{SingerId: n + 1}, // not found
+		tx := client.Single()
+		actual, err := musicdb.Query(tx).BatchGetSingersRows(fx.Ctx, musicdb.BatchGetSingersRowsQuery{
+			Keys: []musicdb.SingersKey{
+				singersAndAlbums[1].Key(),
+				singersAndAlbums[3].Key(),
+				singersAndAlbums[5].Key(),
+				{SingerId: n + 1}, // not found
+			},
+			Albums: true,
```

Comment:
Here's the key update to the API - the method for querying non-interleaved and interleaved results is the same, and there are flags on the query struct for including specific interleaved tables. If no interleaved tables are included, a more efficient query will be constructed that does not use joins at all, that uses the Read APIs instead of the Query APIs if possible.